drools-ansible-rulebook-integration-api/src/test/java/org/drools/ansible/rulebook/integration/api/toexecmodel/MapConditionToPatternSelectTest.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+package org.drools.ansible.rulebook.integration.api.toexecmodel;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.drools.ansible.rulebook.integration.api.domain.conditions.MapCondition;
+import org.drools.core.facttemplates.Event;
+import org.drools.model.functions.Predicate1;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MapConditionToPatternSelectTest extends ToPatternTestBase {
+
+    @Test
+    public void selectExpression() throws Exception {
+        // Create MapCondition
+        LinkedHashMap<Object, Object> lhsValueMap = createLhsForEventField("levels");
+        LinkedHashMap<Object, Object> rhsValueMap = createRhsWithOperatorAndValue("String", ">", "Integer", 25);
+        LinkedHashMap<Object, Object> selectExpression = createSelectExpression(lhsValueMap, rhsValueMap);
+        LinkedHashMap<Object, Object> rootMap = createAllCondition(selectExpression);
+        MapCondition mapCondition = new MapCondition(rootMap);
+
+        // toPattern and extract its predicate
+        Predicate1.Impl predicate = toPatternAndGetFirstPredicate(mapCondition);
+
+        Event event1 = createEvent("levels", 25);
+        assertThat(predicate.test(event1)).isFalse();
+
+        Event event2 = createEvent("levels", 26);
+        assertThat(predicate.test(event2)).isTrue();
+
+        Event event3 = createEvent("levels", 10, 25, 26);
+        assertThat(predicate.test(event3)).isTrue();
+    }
+
+    @Test
+    public void selectNotExpression() throws Exception {
+        // Create MapCondition
+        LinkedHashMap<Object, Object> lhsValueMap = createLhsForEventField("levels");
+        LinkedHashMap<Object, Object> rhsValueMap = createRhsWithOperatorAndValue("String", ">", "Integer", 25);
+        LinkedHashMap<Object, Object> selectNotExpression = createSelectNotExpression(lhsValueMap, rhsValueMap);
+        LinkedHashMap<Object, Object> rootMap = createAllCondition(selectNotExpression);
+        MapCondition mapCondition = new MapCondition(rootMap);
+
+        // toPattern and extract its predicate
+        Predicate1.Impl predicate = toPatternAndGetFirstPredicate(mapCondition);
+
+        Event event1 = createEvent("levels", 25);
+        assertThat(predicate.test(event1)).isTrue();
+
+        Event event2 = createEvent("levels", 26);
+        assertThat(predicate.test(event2)).isFalse();
+
+        Event event3 = createEvent("levels", 10, 25, 26);
+        assertThat(predicate.test(event2)).isFalse();
+    }
+
+    @Test
+    public void selectExpressionWithRegex() throws Exception {
+        // Create MapCondition
+        LinkedHashMap<Object, Object> lhsValueMap = createLhsForEventField("addresses");
+        LinkedHashMap<Object, Object> rhsValueMap = createRhsWithOperatorAndValue("String", "regex", "String", "Main St");
```

Comment:
We should probably put all supported operator types into `String` constants or even better into an `enum`.